specs/FileTypePolicy.md

```diff
@@ -0,0 +1,225 @@
+File Type Policy API
+===
+
+# Background
+When saving a file with original SaveFilePicker, a security alert might be
+prompted, because the browser applies the [file type policies](https://learn.microsoft.com/en-us/deployedge/microsoft-edge-security-downloads-interruptions#file-types-requiring-a-gesture)
+to protect end users. However, in a WebView2 build App, when end users try
+to save a file with a certain file extension, they usually can trust the
+host App, domain and file extension. So, we provide the App developers the
+File Type Policy API to manage the file type policies dynamically.
+
+We'd appreciate your feedback.
+
+# Description
+
+We proposed the `CoreWebView2.FileTypePolicyChecking` event. You can register
```

Comment:
Would this fall into our requested pattern? Something like `SaveFileSecurityDialogRequested`? My mental model is that `*Requested` is roughly that I can choose to do the default thing or override with my own choices and/or UI.

We only use "Show" in `ShowPrintUI`, and "Check" in `IsReputationCheckingRequired`.

Reply:
Yeah deciding if the event is about the security check or the dialog I lean towards security check since the event will be raised even in cases where browser won't show dialog.

Between download and save, neither is perfect but download can refer to things like sub-downloads that don't even go through the download manager and aren't saved to disk in any long term way. Save seems closer to what we're going for.

If we can do something more specific than request and avoid request in the name that's better since it has a specific meaning in WinRT that isn't how we'd be using it.

That's why I like something like:
* CheckingSaveFileSecurity
* SaveFileSecurityCheckStarting